src/main/java/com/monumental/services/ModelService.java

```diff
@@ -397,4 +407,36 @@ private void initializeLazyLoadedCollections(List<T> records) {
             }
         }
     }
+
+    Session openSession() {
+        return this.sessionFactoryService.getFactory().openSession();
+    }
+
+    public EntityManager getEntityManager() {
+        return this.entityManagerFactory.createEntityManager();
+    }
+
+    public CriteriaBuilder getCriteriaBuilder() {
+        return this.getEntityManager().getCriteriaBuilder();
+    }
+
+    public CriteriaQuery<T> createCriteriaQuery() {
+        return this.createCriteriaQuery(this.getCriteriaBuilder());
+    }
+
+    public CriteriaQuery<T> createCriteriaQuery(CriteriaBuilder builder) {
+        return this.createCriteriaQuery(builder, true);
+    }
+
+    public CriteriaQuery<T> createCriteriaQuery(CriteriaBuilder builder, Boolean setRoot) {
+        CriteriaQuery<T> query = builder.createQuery(this.getModelClass());
+        if (setRoot) this.createRoot(query);
+        return query;
+    }
+
+    public Root<T> createRoot(CriteriaQuery<T> query) {
+        Root<T> root = query.from(this.getModelClass());
+        query.select(root);
+        return root;
+    }
```

Comment:
These are basically all provided to make it easier to use the `CriteriaBuilder`, so that you don't necessarily have to do every step of the process every time you want to use it, but if you need any of the intermediary objects (such as `Root`) you are able to do so and then specify it to the later steps.

Usually you should just be able to call `createCriteriaQuery()`, set your clauses, and query it.